lib/gooddata/bricks/middleware/aws_middleware.rb

```diff
@@ -13,20 +13,51 @@ class AWSMiddleware < Bricks::Middleware
       def call(params)
         params = params.to_hash
         if params.key?('aws_client')
-          GoodData.logger.info('Setting up AWS-S3 connection')
+          GoodData.logger.warn('The input_source: aws_client is predicated. Please use s3_client instead.')
           raise 'Unable to connect to AWS. Parameter "aws_client" seems to be empty' unless params['aws_client']
           raise 'Unable to connect to AWS. Parameter "access_key_id" is missing' if params['aws_client']['access_key_id'].blank?
           raise 'Unable to connect to AWS. Parameter "secret_access_key" is missing' if params['aws_client']['secret_access_key'].blank?
-          params['aws_client'] = rewrite_for_aws_sdk_v2(params['aws_client'])
-          symbolized_config = GoodData::Helpers.symbolize_keys(params['aws_client'])
-          s3 = Aws::S3::Resource.new(symbolized_config)
-          params['aws_client']['s3_client'] = s3
+
+          params = create_s3_client(params)
+        elsif params.key?('s3_client')
+          params = rewrite_aws_params(params)
+          params = create_s3_client(params)
         end
         @app.call(params)
       end
 
       private
 
+      def rewrite_aws_params(params)
+        raise 'Unable to connect to AWS. Parameter "s3_client" seems to be empty' unless params['s3_client']
+        raise 'Unable to connect to AWS. Parameter "accessKey" is missing' if params['s3_client']['accessKey'].blank?
+        raise 'Unable to connect to AWS. Parameter "secretKey" is missing' if params['s3_client']['secretKey'].blank?
+
+        params['aws_client'] = {}
```

Comment:
aws_client is deprecated, we should rewrite it opposite to s3_client instea.

Reply:
Updated to use s3_client instead. Thanks